tools/pathwaymatcher/pathwaymatcher.xml

```diff
@@ -0,0 +1,332 @@
+        <regex match="Could not create the Java virtual machine" level="fatal" description="JVM Error"/>
+    </stdio>
+    <command>
+<![CDATA[
+        #from datetime import datetime
+        #import json
+        #import os
+        #set $exp_str = "Galaxy_Experiment_%s" % datetime.now().strftime("%Y%m%d%H%M%s")
+        #set $samp_str = "Sample_%s" % datetime.now().strftime("%Y%m%d%H%M%s")
+        #set $temp_stderr = "pathwaym_stderr"
+        #set $bin_dir = "bin"
+
+        mkdir output;
+        cwd=`pwd`;
+        export HOME=\$cwd;
+
+        #####################
+        ## Pathway Matcher ##
+        #####################
+        (pathwaymatcher src.main.java.no.uib.pap.pathwaymatcher.PathwayMatcher
+
+            ## GENETIC VARIANTS
+
+            #if $genetic_variants.input_rsid:
+                -t rsid -i "${genetic_variants.input_rsid}"
+            #end if
+
+            #if $genetic_variants.input_chrbp:
+                -t chrbp -i "${genetic_variants.input_chrbp}"
+            #end if
+
+            #if $genetic_variants.input_vcf:
+                -t vcf -i "${genetic_variants.input_vcf}"
+            #end if
+
+
+            ## GENES
+
+            #if $genes.input_gene:
+                -t gene -i "${genes.input_gene}"
+            #end if
+
+            ## PEPTIDES
+
+##            #if $peptides.input_peptide:
+##                -t peptide -i "${peptides.input_peptide}"
+##            #end if
+
+##            #if $peptides.input_modifiedpeptide:
+##                -t modifiedpeptide -i "${peptides.input_modifiedpeptide}"
+##            #end if
+
+##            #if $peptides.ptm_range:
+##                -r "${peptides.ptm_range}"
+##            #end if
+
+
+            ## PROTEINS
+
+            #if $proteins.input_uniprot:
+                -t uniprot -i "${proteins.input_uniprot}"
+            #end if
+
+            #if $proteins.input_ensembl:
+                -t ensembl -i "${proteins.input_ensembl}"
+            #end if
+
+
+            ## PROTEOFORMS
+
+            #if $proteoforms.input_proteoforms:
+                #if $proteoforms.proteoform_match_criteria:
+                  -t proteoform -m "${proteoforms.proteoform_match_criteria}" -i "${proteoforms.input_proteoforms}"
+                #else:
+                  -t proteoform -i "${proteoforms.input_proteoforms}"
+                #end if
+            #end if
+
+
+            ## OUTPUT OPTIONS
+
+            #if $output_options.proteoform_match_criteria:
+                -tlp
+            #end if
+
+            #set $output_graphs_list = str($output_options.output_graphs).split(',')
+
+            #if 'gg' in $output_graphs_list:
+                -gg
+            #end if
+
+            #if 'gu' in $output_graphs_list:
+                -gu
+            #end if
+
+            #if 'gp' in $output_graphs_list:
+                -gp
+            #end if
+
+        2>> $temp_stderr);
+
+        ## We create a zip file to contain graphs files.
```

Comment:
Why not creating a collection with all tsv files?

Reply:
I thought about it but, when you generate at least one graphic, you obtain 3 files; with the 3 graphs chosen, you get 9 different files (all of them related in groups of 3). You may be only interested in some specific ones (ie:internalEdges of genes, proteins and proteoforms) but still you would fill your history with them all (9), so in order to avoid this I think it's useful to have only one compressed file (which will also be much lighter to download with big datasets) with everything into it, also independently of the number of graphs you chose.

Reply:
I don't get this argument. If you just want to have one triple, then you need to download the entire zip, extract it and search for the files. In collections you just open them, browse and click on the file your are interested in. A collection is just one dataset in our history, so it does not pollute your history. Downloading should be possible as entire collection. I do agree its more complicated than a zip hack, but I think from a UX point of view its the correct things to do.

It your tool, I do not want to push you away, sorry if this is the impression :( - just want to provide hopefully useful tips.

Reply:
Don't worry, I really appreciate your advice. It seems I was thinking on collections wrongly, as if they occupied many rows in the history. I will check and try to use them :-)